### PR TITLE
📖 Update Versioning notes

### PR DESCRIPTION
**Pull request by furkatgofurov7**
**What this PR does / why we need it**:

It fixes:
-  dead link reference, Prow link reference
-  CNCF CLA check for PR authors' notes since it is not required

### CODE REVIEW

**Review by lentzi90 — APPROVED** [review]

**Review by Sunnatillo — COMMENTED** [review]
LGTM

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

File: VERSIONING.md [hunk 1]
```diff
@@ -73,28 +73,27 @@ a:
...
+
 ### Commands and Workflow
 
 Cluster API Provider Metal3 follows the standard Kubernetes workflow: any PR
-needs `lgtm` and `approved` labels, PRs authors must have signed the CNCF CLA,
-and PRs must pass the tests before being merged.  See [the contributor
-docs](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#the-testing-and-merge-workflow)
-for more info.
+needs `lgtm` and `approved` labels, and PRs must pass the tests before being merged.
+See [the contributor docs](https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#the-testing-and-merge-workflow) for more info.
```
Comment (kashifest) [inline VERSIONING.md:90]: Shall we keep CNCF CLA and ask for it from authors rather ?

**Review by kashifest — APPROVED** [review]
One comment inline

File: VERSIONING.md (hunk 1 above)
Comment (furkatgofurov7) [reply VERSIONING.md:90]: Before change, it says `PRs authors must have signed the CNCF CLA,` which makes me think we have a mechanism to check if it is not the case, so I removed it to make it clear. Can you elaborate more what you mean by ask from authors?

Comment (kashifest) [reply VERSIONING.md:90]: to have CNCF CLA

Comment (kashifest) [reply VERSIONING.md:90]: We have to check if we can check existence of CNCF CLA in prow, for now I am ok to remove it and we can check it later

Comment (furkatgofurov7) [reply VERSIONING.md:90]: Sure, I will look into putting CNCF CLA check in prow for this repo at least.

**Pull request merged**